includes/admin/marketplace-suggestions/class-wc-marketplace-suggestions.php

```diff
@@ -0,0 +1,167 @@
+			wp_die();
+		}
+
+		$post_data       = wp_unslash( $_POST );
+		$suggestion_slug = sanitize_text_field( $post_data['slug'] );
+		if ( ! $suggestion_slug ) {
+			wp_die();
+		}
+
+		$dismissed_suggestions = self::get_dismissed_suggestions();
+
+		if ( in_array( $suggestion_slug, $dismissed_suggestions, true ) ) {
+			wp_die();
+		}
+
+		$dismissed_suggestions[] = $suggestion_slug;
+		update_user_meta(
+			get_current_user_id(),
+			'wc_marketplace_suggestions_dismissed_suggestions',
+			$dismissed_suggestions
+		);
+
+		wp_die();
+	}
+
+	/**
+	 * Render suggestions containers in products list empty state.
+	 */
+	public static function render_products_list_empty_state() {
+		self::render_suggestions_container( 'products-list-empty-header' );
+		self::render_suggestions_container( 'products-list-empty-body' );
+		self::render_suggestions_container( 'products-list-empty-footer' );
+	}
+
+	/**
+	 * Render a suggestions container element, with the specified context.
+	 *
+	 * @param string $context Suggestion context name (rendered as a css class).
+	 */
+	public static function render_suggestions_container( $context ) {
+		include 'templates/container.php';
+	}
+
+	/**
+	 * Should suggestions be displayed?
+	 *
+	 * User can disabled all suggestions via option.
+	 * We also currently only support English suggestions.
+	 *
+	 * @return bool
+	 */
+	public static function allow_suggestions() {
+		$option = get_option( 'woocommerce_allow_marketplace_suggestions', 'yes' );
+		$allow  = ( 'yes' === $option || 'true' === $option || 1 === $option );
+
+		$locale             = get_locale();
+		$suggestion_locales = array(
+			'en_AU',
+			'en_CA',
+			'en_GB',
+			'en_NZ',
+			'en_US',
+			'en_ZA',
+		);
+		$language_supported = in_array( $locale, $suggestion_locales, true );
+
+		return ( $allow && $language_supported );
+	}
+
+	/**
+	 * Pull suggestion data from remote endpoint & cache in a transient.
+	 *
+	 * @return array of json API data
+	 */
+	private static function get_suggestions_api_data() {
+		$suggestion_data = get_transient( 'wc_marketplace_suggestions' );
+		if ( false !== $suggestion_data ) {
+			return $suggestion_data;
+		}
+
+		$suggestion_data_url = 'https://d3t0oesq8995hv.cloudfront.net/add-ons/marketplace-suggestions.json';
+		$raw_suggestions     = wp_remote_get(
+			$suggestion_data_url,
+			array( 'user-agent' => 'WooCommerce Marketplace Suggestions' )
+		);
+
+		// Parse the data to check for any errors.
+		// If it's valid, store structure in transient.
+		if ( ! is_wp_error( $raw_suggestions ) ) {
+			$suggestions = json_decode( wp_remote_retrieve_body( $raw_suggestions ) );
+			if ( $suggestions && is_array( $suggestions ) ) {
+				set_transient( 'wc_marketplace_suggestions', $suggestions, WEEK_IN_SECONDS );
+				return $suggestions;
+			}
+		}
+
+		// Cache empty suggestions data to reduce requests if there are any issues with API.
+		set_transient( 'wc_marketplace_suggestions', '[]', DAY_IN_SECONDS );
```

Comment:
Should this be a real array perhaps? it's returned on line 142 so should probably not be a string.

Reply:
163af89ab3a1cd56bdc862ada6b64086a9db7aa3